### PR TITLE
Added compatibility for builds with Java 11

### PR DESCRIPTION
**Pull request by todorinskiz**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the changelog.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -38,6 +38,14 @@ allprojects {
     apply plugin: 'idea'
     apply plugin: 'eclipse'
 
+    tasks.withType(JavaCompile) {
```
Comment (vierbergenlars) [inline build.gradle]: It's better to set `sourceCompatibility` directly on the project, and don't set `targetCompatibility`, as it defaults to be the same as `sourceCompatibility`.

_Why?_
Because setting it on the project will result in the correct attributes being applied to outgoing configurations (used for for dependencies between projects). The attributes mark which version of Java the jar is compatible with, and this is derived from `sourceCompatibility` as set on the project.

File: alfresco-dynamic-extensions-repo/build.gradle [hunk 2]
```diff
@@ -27,7 +27,7 @@ subprojects {
         ampLib(project(":annotations-runtime:scheduler-quartz-1")) { transitive = false }
         ampLib(project(":annotations-runtime:scheduler-quartz-2")) { transitive = false }
 
-        ampLib(project(":webscripts")) { transitive = false }
```
Comment (kerkhofsd) [inline alfresco-dynamic-extensions-repo/build.gradle]: How is this related? What does is change / fix?

File: build.gradle (hunk 1 above)
Comment (kerkhofsd) [reply build.gradle]:
```
    project.pluginManager.withPlugin('java', {

        tasks.withType(JavaCompile) {
            sourceCompatibility = 1.8
        }
    })
```

@vierbergenlars what do you think of doing it like this?

Comment (kerkhofsd) [reply build.gradle]: wait, that still sets the `sourceCompatibility` on the tasks. 

So it should be this:
```
    project.pluginManager.withPlugin('java', {
       sourceCompatibility = 1.8
    })
```

?

Comment (vierbergenlars) [reply build.gradle]: It should be configured directly on the project, not on a `JavaCompile` task, so:

```groovy
project.pluginManager.withPlugin('java-base') {
    project.sourceCompatibility = 1.8
}
```

Comment (kerkhofsd) [reply build.gradle]: I assume both `java` and `java-library` apply `java-base`?

Comment (vierbergenlars) [reply build.gradle]: Yes, that is correct. And it is the `java-base` plugin that adds the `sourceCompatibility` extension to a project, so that's why you use that plugin in `withPlugin`

File: alfresco-dynamic-extensions-repo/build.gradle (hunk 2 above)
Comment (todorinskiz) [reply alfresco-dynamic-extensions-repo/build.gradle]: Added comments, see on the right

File: build.gradle (hunk 1 above)
Comment (todorinskiz) [reply build.gradle]: Done, could you please review the changes? :)

File: alfresco-dynamic-extensions-repo/build.gradle [hunk 3]
```diff
@@ -27,7 +27,9 @@ subprojects {
...
+        // Transitive dependencies are required to retain javax.annotations dependency
```
Comment (vierbergenlars) [inline alfresco-dynamic-extensions-repo/build.gradle]: It is not dynamic-extension's job to ship `javax.annotations` for all of Alfresco, and this may result in multiple identical classes on the classpath, and all problems that arise from it.

Comment (todorinskiz) [reply alfresco-dynamic-extensions-repo/build.gradle]: @vierbergenlars can you propose a better solution?

Comment (kerkhofsd) [reply alfresco-dynamic-extensions-repo/build.gradle]: @todorinskiz this `javax.annotations` dependency is provided by Alfresco?

Comment (vierbergenlars) [reply alfresco-dynamic-extensions-repo/build.gradle]: I would be very surprised if nothing of Alfresco or Spring uses these annotations, and already has it as a dependency.

File: build.gradle [hunk 4]
```diff
@@ -38,6 +38,10 @@ allprojects {
     apply plugin: 'idea'
     apply plugin: 'eclipse'
 
+    project.pluginManager.withPlugin('java-base') {
```
Comment (kerkhofsd) [inline build.gradle]: can we add 
```
        tasks.withType(JavaCompile) {
            options.encoding = "UTF-8"
            options.compilerArgs << '-Xlint:unchecked'
        }
```
here and remove the redundant code block from the `subprojects` configuration?

File: control-panel/build.gradle [hunk 5]
```diff
@@ -48,6 +48,7 @@ jar {
             'Alfresco-Spring-Configuration': 'com.github.dynamicextensionsalfresco.controlpanel',
             'Spring-Context': ';publish-context:=false',
             'Import-Package': '*, ' +
+                    'javax.annotation;version=!, ' +
```
Comment (kerkhofsd) [inline control-panel/build.gradle]: The '*' should move to the end, otherwise `javax.annotation` is twice in the `Import-Package` header.

File: build.gradle (hunk 4 above)
Comment (vierbergenlars) [reply build.gradle]: That will work yes, then everything is a bit more together
